enterprise/internal/campaigns/rewirer/rewirer.go

```diff
@@ -199,16 +199,42 @@ func (r *ChangesetRewirer) closeChangeset(changeset *campaigns.Changeset) {
 	}
 }
 
+func attachToCampaign(changeset *campaigns.Changeset, campaignID int64) {
+	for i := range changeset.Campaigns {
+		if changeset.Campaigns[i].CampaignID == campaignID {
+			changeset.Campaigns[i].Detach = false
+			return
+		}
+	}
+	changeset.Campaigns = append(changeset.Campaigns, campaigns.CampaignAssoc{CampaignID: campaignID})
+}
+
+// ErrRepoNotSupported is thrown by the rewirer when it encounters a mapping
+// targetting a repo on a code host that's not supported by campaigns.
+type ErrRepoNotSupported struct {
+	ServiceType string
+	RepoName    string
+}
+
+func (e ErrRepoNotSupported) Error() string {
+	return fmt.Sprintf(
+		"External service type %s of repository %q is currently not supported for use with campaigns",
```

Comment:
```suggestion
		"Code host type %s of repository %q is currently not supported for use with campaigns",
```

If we're touching it, we might as well update the wording :)